### DISCUSSION TITLE
Support configurability of authentication for Gitlab

### DISCUSSION THREAD

**Discussion by k3vonk** [renovatebot/renovate#23630] [Suggest an Idea]
### Tell us more.

Gitlab supports 2 core flows of [authenticating requests](https://docs.gitlab.com/ee/api/rest/index.html#personalprojectgroup-access-tokens). Either through `PRIVATE-TOKEN` or `Authorization: Bearer`.

The current solution defaults to `Bearer` (https://github.com/renovatebot/renovate/blob/b02d4ac49216fdbda577c364ac8b90f497ef6f12/lib/util/http/auth.ts#L54C1-L58). However, there are specific scenarios where we might overwrite the `Authorization` header e.g. using an AWS Sigv4 Proxy.

It would be great to allow the user to have the option to set up requests through `PRIVATE-TOKEN`

**Comment by rarkins** [comment]
Could you describe this scenario more?
